android/libraries/rib-base/src/main/java/com/badoo/ribs/core/Node.kt

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright (C) 2017. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.badoo.ribs.core
+
+import android.os.Bundle
+import android.os.Parcelable
+import android.support.annotation.CallSuper
+import android.support.annotation.MainThread
+import android.support.annotation.VisibleForTesting
+import android.util.SparseArray
+import android.view.ViewGroup
+import com.badoo.ribs.core.view.RibView
+import com.badoo.ribs.core.view.ViewFactory
+import com.badoo.ribs.core.requestcode.RequestCodeRegistry
+import com.uber.rib.util.RibRefWatcher
+import java.util.UUID
+import java.util.concurrent.CopyOnWriteArrayList
+
+/**
+ * Responsible for handling the addition and removal of child nodes.
+ **/
+open class Node<V : RibView>(
+    val forClass: Class<*>,
+    private val viewFactory: ViewFactory<V>?,
+    private val router: Router<*, V>,
+    private val interactor: Interactor<*, V>,
+    private val ribRefWatcher: RibRefWatcher = RibRefWatcher.getInstance()
+) {
+    companion object {
+        @VisibleForTesting internal val KEY_CHILD_NODES = "node.children"
```

Comment:
I think you are quite safe event with just `internal`, as it limits the visibility scope to the compilation module